lightningnetwork/lnd
# lnd: compute FundingExpiryBlocks in PendingOpenChannel message

## Change Description

This adds the `FundingExpiryBlocks` field to the `PendingOpenChannel` message. The `FundingExpiryBlocks` field is computed from the broadcast height of the funding transaction, the current height of the chain and `MaxWaitNumBlocksFundingConf`. This is visible via the RPC and `lncli pendingchannels`.

Resolves #7434

## Steps to Test

Added an integration test (`testFundingExpiryBlocksOnPending`) that opens a channel between Alice and Bob, invokes the RPC "pendingchannels", checks for the `FundingExpiryBlocks` field in the result message, mines a block, and then loops checking that the value decreases as expected until the funding transaction is completed.

> $ make itest icase=funding_expiry_blocks_on_pending

Also manually tested the results of `lncli pendingchannels`:
- Created a simnet between Alice and Bob with `lncli openchannel`.
- Check that `lncli pendingchannels` has `"funding_expiry_blocks": 2016`.
- Mine a block with `btcd generate 1`.
- Check that `lncli pendingchannels` has `"funding_expiry_blocks": 2015`. The value decreased by one as expected.
- Mine two more blocks and the channel moves from pending to open and does not show up with `lncli pendingchannels`, but rather does show up with `lncli listchannels`.

## Pull Request Checklist
### Testing

### Code Style and Documentation

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.
